src/compile/axis/rules.ts

```diff
@@ -53,6 +53,24 @@ export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, ch
   return undefined;
 }
 
+export function minMaxExtent(specifiedExtent: number, isGridAxis: boolean, scaleType: ScaleType) {
+  if (!isGridAxis) {
+    // Only apply extent to non-grid axes as grid axes do not have labels and title.
+    // Otherwise, we would add unnecessary extent to faceted plots.
+    if (specifiedExtent) {
+      return specifiedExtent;
+    }
+
+    // For quantitative scale, set extent to 25 by default to avoid jumpy axis title
+    // (Fix https://github.com/vega/vega-lite/issues/2282)
+    if (getScaleCategory(scaleType) === 'numeric') {
+      return 25;
+    }
+  }
+
+  return undefined;
+}
```

Comment:
> Should minExtent and maxExtent be added to AXIS_PROPERTY_TYPE in axis.ts?

Yes.  Great catch.  I actually added `AXIS_PROPERTY_TYPE` because I think I need it for `minExtent` and `maxExtent` but I stash and swap commit order and forgot to bring it back :p